database/database_test.go

```diff
@@ -234,3 +236,423 @@ func TestStatesWithStatus(t *testing.T) {
+	var statuses []string
+	statusReport := func(format string, args ...interface{}) {
+		formatted := fmt.Sprintf(format, args...)
+		statuses = append(statuses, formatted)
+	}
+	for _, tt := range getAndDeleteTests {
+		err := testDB.SetIneligibleDomainStates(
+			tt.wantStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+	}
+	// delete domains from the database
+	for _, tr := range getAndDeleteTests {
+		err = testDB.DeleteIneligibleDomainStates(tr.domainNames)
+		if err != nil {
+			t.Errorf("%s", err)
+		}
+	}
+
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
+
+}
+
+// setIneligibleDomainTests is a struct that is used in testing SetIneligibleDomainStates
+var setGetDeleteDuplicateIneligibleDomainTests = []struct {
+	description       string
+	domainNames       []string
+	domainStates      []IneligibleDomainState
+	wantStatusReports []string
+}{
+	{
+		"err: formatting error",
+		[]string{"gmail.test", "gmail.test"},
+		[]IneligibleDomainState{{Name: "gmail.test", Policy: "bulk-18-week", Scans: []Scan{ 
+			{
+				ScanTime: time.Unix(1234,54324),
+				Issues: []hstspreload.Issues{
+					{
+						Errors: []hstspreload.Issue{
+							{
+								Code: "format_error", 
+								Summary: "Formatting error",
+								Message: "Please fix the format in your code",
+							},
+						},
+					},
+				},
+			},
+		}},
+		{Name: "gmail.test", Policy: "bulk-18-week", Scans: []Scan{ 
+			{
+				ScanTime: time.Unix(1234,54324),
+				Issues: []hstspreload.Issues{
+					{
+						Errors: []hstspreload.Issue{
+							{
+								Code: "format_error", 
+								Summary: "Formatting error",
+								Message: "Please fix the format in your code",
+							},
+						},
+					},
+				},
+			},
+		}}},
+		[]string{"done. \n"},
+	},
+	{
+		"err: domain does not exist",
+		[]string{"youtube.test", "youtube.test"},
+		[]IneligibleDomainState{{Name: "youtube.test", Policy: "bulk-1-year", Scans: []Scan{ 
+			{
+				ScanTime: time.Unix(1234,54324),
+				Issues: []hstspreload.Issues{
+					{
+						Errors: []hstspreload.Issue{
+							{
+								Code: "formatting error", 
+								Summary: "missing end of page line",
+								Message: "add a line at the end of the page",
+							},
+						},
+					},
+				},
+			},
+		}},
+		{Name: "youtube.test", Policy: "bulk-1-year", Scans: []Scan{ 
+			{
+				ScanTime: time.Unix(1234,54324),
+				Issues: []hstspreload.Issues{
+					{
+						Errors: []hstspreload.Issue{
+							{
+								Code: "formatting error", 
+								Summary: "missing end of page line",
+								Message: "add a line at the end of the page",
+							},
+						},
+					},
+				},
+			},
+		}},},
+		[]string{"done. \n"},
+	},
+}
+
+// TestSetGetDeleteDuplicateIneligibleDomainStates tests adding, getting, and deleting duplicate 
+// domain states from the database
+func TestSetGetDeleteDuplicateIneligibleDomainStates(t *testing.T) {
```

Comment:
This test is testing too many different things. I think it's testing some subset of the following, each of which should be a separate test:

- a call to `SetIneligibleDomainStates` with duplicate entries
- a call to `GetIneligibleDomainStates` with duplicate names
- a call to `GetIneligibleDomainStates` with a name that doesn't exist
- a call to `DeleteIneligibleDomainStates` with duplicate names
- a call to `DeleteIneligibleDomainStates` with a name that doesn't exist